plugin/src/main/java/org/opensearch/ml/action/tasks/DeleteTaskTransportAction.java

```diff
@@ -35,65 +51,156 @@
-        GetRequest getRequest = new GetRequest(ML_TASK_INDEX).id(taskId);
+        String tenantId = mlTaskDeleteRequest.getTenantId();
+
+        if (!TenantAwareHelper.validateTenantId(mlFeatureEnabledSetting, tenantId, actionListener)) {
+            return;
+        }
+
+        FetchSourceContext fetchSourceContext = new FetchSourceContext(true, Strings.EMPTY_ARRAY, Strings.EMPTY_ARRAY);
+        GetDataObjectRequest getDataObjectRequest = GetDataObjectRequest
+            .builder()
+            .index(ML_TASK_INDEX)
+            .id(taskId)
+            .fetchSourceContext(fetchSourceContext)
+            .build();
 
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-            ActionListener<DeleteResponse> wrappedListener = ActionListener.runBefore(actionListener, () -> context.restore());
-            client.get(getRequest, ActionListener.wrap(r -> {
-
-                if (r != null && r.isExists()) {
-                    try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
-                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
-                        MLTask mlTask = MLTask.parse(parser);
-                        MLTaskState mlTaskState = mlTask.getState();
-                        if (mlTaskState.equals(MLTaskState.RUNNING)) {
-                            actionListener.onFailure(new Exception("Task cannot be deleted in running state. Try after sometime"));
-                        } else {
-                            DeleteRequest deleteRequest = new DeleteRequest(ML_TASK_INDEX, taskId).setRefreshPolicy(IMMEDIATE);
-                            client.delete(deleteRequest, new ActionListener<DeleteResponse>() {
-                                @Override
-                                public void onResponse(DeleteResponse deleteResponse) {
-                                    log.debug("Completed Delete Task Request, task id:{} deleted", taskId);
-                                    wrappedListener.onResponse(deleteResponse);
-                                }
-
-                                @Override
-                                public void onFailure(Exception e) {
-                                    log.error("Failed to delete ML Task " + taskId, e);
-                                    wrappedListener.onFailure(e);
-                                }
-                            });
-                        }
-                    } catch (Exception e) {
-                        log.error("Failed to parse ML task " + taskId, e);
-                        wrappedListener.onFailure(e);
+            ActionListener<DeleteResponse> wrappedListener = ActionListener.runBefore(actionListener, context::restore);
+
+            sdkClient
+                .getDataObjectAsync(getDataObjectRequest, client.threadPool().executor(GENERAL_THREAD_POOL))
```

Comment:
I thought we were avoiding using the thread pool here?

Reply:
missed that, addressing it.